csrc/adagrad/custom_cuda_kernel.cu

```diff
@@ -0,0 +1,41 @@
+

```

Comment:
Can we use the same cuda kernels as for cpu-adam rather than duplicating them?

Reply:
Sure, I will refactor this and below one.